### ISSUE TITLE
Nested composites with cyclic dependencies cause an error

### ISSUE THREAD

**Issue by nddipiazza** [gradle/gradle#6147]
Version: 4.10.nightly

Let's say you have the following dependencies between composite builds:

```
    projectA --compile--> 
      projectB --testcompile--> 
        projectC --compile--> 
          projectA
```

This does not create a cycle due to the testcompile dependency between projectB and projectC.

Now let's say each of these projects is connected with composite build `includedBuild` such that

```
projectA
   includeBuild '../projectB'
projectB
   includeBuild '../projectC'
projectC
   includeBuild '../projectA'
```

Now if you try to build `cd projectA; ./gradlew build` 

This will cause the error: 

```
Included build in /Users/nicholasdipiazza/projectA has the same root project name 'projectA' as the main build.
```

This is because you are building root project `projectA` but then projectC includes `projectA`.

This should just realize that you already included that build and ignore it. 

The work around is to build this project from a different project root, then it works fine.

**Comment by calvertdw** [comment]
> This should just realize that you already included that build and ignore it.

I agree and this is what I do in my plugin:

```
   val transitiveIncludedBuilds = TreeSet<IHMCBuildProperties>()
   ...
   transitiveIncludedBuilds.add(propertiesFromKebabCasedName(artifactName))
```

https://github.com/ihmcrobotics/ihmc-build/blob/8d038dd477043862ae50ccaf44987afc7616e15e/src/main/kotlin/us/ihmc/build/IHMCCompositeBuildAssembler.kt#L27

**Comment by sirinath** [comment]
With the new composite script plugins, there is always a chance to double including a plugin which needs to be handled.

**Comment by aamiel16** [comment]
Any plans to support this in the near future?

**Comment by calvertdw** [comment]
This would be a nice feature to have!

**Comment by ov7a** [comment]
This issue was fixed in 6.8.

https://docs.gradle.org/6.8/release-notes.html
